### PR TITLE
Add marker-based pagination for getting users

### PR DESCRIPTION
**Pull request by sujaygarlanka**
- Updated PagingIterator by adding getter method getNextMarker(). The next marker can currently be retrieved by getting the nextValue class variable on the iterator since it is public.
- Updated documentation on how to use either offset-based or marker-based pagination for the methods that support both

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sujaygarlanka has signed the CLA. Thanks for the pull request!

File: docs/enterprise.md [hunk 1]
```diff
@@ -18,11 +18,11 @@ Get Enterprise Users
 
 Get a list of users in the current enterprise by calling the
 [`enterprise.getUsers(options, callback)`](http://opensource.box.com/box-node-sdk/jsdoc/Enterprise.html#getUsers)
-method.
+method. This method supports offset-based pagination and marker-based pagintation. To use offset-based pagination, do not pass in the `usemarker` parameter or set it to `false`. To use marker-based pagination, pass in the `usemarker` parameter as `true`. Use the `fields` option to specify the desired fields, and `limit` and (`offset` or `marker`) to control result set paging. Requesting information for only the fields you need can improve performance and reduce the size of the network request.
```
Comment (PJSimon) [inline docs/enterprise.md]:
```suggestion
method. This method supports offset-based pagination and marker-based pagintation. To use offset-based pagination, do not pass in the `usemarker` parameter or set it to `false`. To use marker-based pagination, pass in the `usemarker` parameter as `true`. Use the `fields` option to specify the desired response fields, and `limit` (along with `offset` or `marker`) to control result set paging. Requesting information for only the fields you need can improve performance by reducing the size of the network response.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## Upcoming Release
+
+- Added a getter method (getNextMarker()) for the next marker to PagingIterator
```
Comment (PJSimon) [inline CHANGELOG.md]: Thanks!
I hope I remember to update the CHANGELOG on my next PR.

File: docs/folders.md [hunk 3]
```diff
@@ -119,14 +119,14 @@ Get a Folder's Items
 
 Folder items can be retrieved by calling the
 [`folders.getItems(folderID, options, callback)`](http://opensource.box.com/box-node-sdk/jsdoc/Folders.html#getItems)
-method. Use the `fields` option to specify the desired fields, and `limit` and `offset` to control result set paging.
-Requesting information for only the fields you need can improve performance and reduce the size of the network request.
+method. This method supports offset-based pagination and marker-based pagination. To use offset-based pagination, do not pass in the `usemarker` parameter or set it to `false`. To use marker-based pagination, pass in the `usemarker` parameter as `true`. Use the `fields` option to specify the desired fields, and `limit` and (`offset` or `marker`) to control result set paging. Requesting information for only the fields you need can improve performance and reduce the size of the network request.
```
Comment (PJSimon) [inline docs/folders.md]:
```suggestion
method. This method supports offset-based pagination and marker-based pagination. To use offset-based pagination, do not pass in the `usemarker` parameter or set it to `false`. To use marker-based pagination, pass in the `usemarker` parameter as `true`. Use the `fields` option to specify the desired fields, and `limit` and (`offset` or `marker`) to control result set paging. Requesting information for only the fields you need can improve performance by reducing the size of the network response.
```

**Pull request merged**